WearSpeakerSample/wear/src/main/java/com/example/android/wearable/speaker/SpeakerScreen.kt

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.speaker
+
+import android.content.res.Configuration
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.EnterTransition
+import androidx.compose.animation.ExitTransition
+import androidx.compose.animation.expandIn
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.animation.shrinkOut
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.datasource.CollectionPreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
+import androidx.wear.compose.material.ExperimentalWearMaterialApi
+import androidx.wear.compose.material.Scaffold
+import androidx.wear.compose.material.TimeText
+
+/**
+ * The composable responsible for displaying the main UI.
+ *
+ * This composable is stateless, and simply displays the state given to it.
+ */
+@OptIn(ExperimentalWearMaterialApi::class)
+@Composable
+fun SpeakerScreen(
+    appState: AppState,
+    isPermissionDenied: Boolean,
+    recordingProgress: Float,
+    onMicClicked: () -> Unit,
+    onPlayClicked: () -> Unit,
+    onMusicClicked: () -> Unit
+) {
+    Scaffold(
+        modifier = Modifier.background(colorResource(id = R.color.black)),
+        timeText = {
+            TimeText()
+        }
+    ) {
+        // Determine the control dashboard state.
+        // This converts the main app state into a control dashboard state for rendering
+        val controlDashboardState by derivedStateOf {
```

Comment:
Extract this out to a function in this file? Adds a bit of noise and doesn't actually help me understand what `SpeakerScreen` is doing